client_test.go

```diff
@@ -389,6 +389,15 @@ func Test_Client_Stop(t *testing.T) {
 		}
 	}
 
+	t.Run("not started", func(t *testing.T) {
+		t.Parallel()
+		client := newTestClient(ctx, t, newTestConfig(t, nil))
+
+		err := client.Stop(ctx)
+		require.Error(t, err)
+		require.Equal(t, "client not started", err.Error())
```

Comment:
another option for checking the error string, would be to expose it as a sentinel error, e.g.
`NotStartedErr`, but I saw that there are three defined in `client.go` but they are not exported, so I opted for this instead

Reply:
Yeah. I think it's okay not to give them a full error type until we're more sure that someone will need it. This case is pretty clearly a misuse problem that'll rarely occur outside of dev and not need normal handling.

Reply:
that was exactly how I found it - I messed up my tests setup and got a panic